pkg/trace/event/processor.go

```diff
@@ -66,24 +67,26 @@ func (p *Processor) Process(root *pb.Span, t *pb.TraceChunk) (numEvents, numExtr
 		}
 
 		numExtracted++
-
-		sampled, epsRate := p.maxEPSSample(span, priority)
-		if !sampled {
-			continue
+		//event analytics tags shouldn't be set on sampled single spans
+		if !(span.Metrics != nil &&
+			span.Metrics[sampler.KeySpanSamplingMechanism] == float64(apmsampling.SpanSamplingRuleMechanism)) {
```

Comment:
```suggestion
		if traceutil.GetSamplingMechanism(span) != apmsampling.SpanSamplingRuleMechanism {
```
Maybe this would make things easier and clearer. You'll have to create the helper.